fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/expressions/LogicalSortExpression.java

```diff
@@ -41,20 +41,32 @@
 @API(API.Status.EXPERIMENTAL)
 public class LogicalSortExpression implements RelationalExpressionWithChildren {
     @Nonnull
-    private final KeyExpressionComparisons sort;
+    private final List<Element> grouping;
+    @Nonnull
+    private final List<Element> sort;
     private final boolean reverse;
     @Nonnull
     private final ExpressionRef<RelationalPlannerExpression> inner;
 
-    public LogicalSortExpression(@Nonnull KeyExpression sort, boolean reverse, @Nonnull RelationalPlannerExpression inner) {
+    public LogicalSortExpression(@Nonnull List<Element> sort, boolean reverse, @Nonnull RelationalPlannerExpression inner) {
         this(sort, reverse, SingleExpressionRef.of(inner));
     }
 
-    public LogicalSortExpression(@Nonnull KeyExpression sort, boolean reverse, @Nonnull ExpressionRef<RelationalPlannerExpression> inner) {
-        this(new KeyExpressionComparisons(sort), reverse, inner);
+    public LogicalSortExpression(@Nonnull List<Element> grouping, @Nonnull List<Element> sort, boolean reverse, @Nonnull RelationalPlannerExpression inner) {
+        this.grouping = grouping;
+        this.sort = sort;
+        this.reverse = reverse;
+        this.inner = SingleExpressionRef.of(inner);
+    }
+
+
+    public LogicalSortExpression(@Nonnull List<Element> sort, boolean reverse, @Nonnull ExpressionRef<RelationalPlannerExpression> inner) {
+        this(Collections.emptyList(), sort, reverse, inner);
     }
 
-    public LogicalSortExpression(@Nonnull KeyExpressionComparisons sort, boolean reverse, @Nonnull ExpressionRef<RelationalPlannerExpression> inner) {
+    public LogicalSortExpression(@Nonnull List<Element> grouping, @Nonnull List<Element> sort, boolean reverse,
+                                 @Nonnull ExpressionRef<RelationalPlannerExpression> inner) {
```

Comment:
I agree that this could use some explanation. We're going to move away from this paradigm entirely in the next batch of changes, so I'm inclined to leave it as-is for now.